pkg/plugin/resource/resource.go

```diff
@@ -0,0 +1,200 @@
+package resource
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"sort"
+	"text/template"
+
+	log "github.com/Sirupsen/logrus"
+	plugin_group "github.com/docker/infrakit/pkg/plugin/group"
+	"github.com/docker/infrakit/pkg/plugin/group/types"
+	"github.com/docker/infrakit/pkg/spi/group"
```

Comment:
See comment below on creating a new SPI for resource.  Ideally this package won't have any import dependencies on the group pkg.   All the specs and types that currently reference the group pkg should just be recreated and put in the resource pkg.  This way, the resource plugin SPI is free to evolve on its own.